perllib/FixMyStreet/Script/UpdateAllReports.pm

```diff
@@ -43,23 +44,33 @@ sub generate {
     $problems = $problems->cursor; # Raw DB cursor for speed
 
     my ( %fixed, %open );
-    my @cols = ( 'id', 'bodies_str', 'state', 'areas', 'cobrand', 'duration', 'age' );
+    my %stats = (
+        fixed => \%fixed,
+        open => \%open,
+    );
+    my @cols = ( 'id', 'bodies_str', 'state', 'areas', 'cobrand', 'category', 'duration', 'age' );
     while ( my @problem = $problems->next ) {
         my %problem = zip @cols, @problem;
-        my @bodies;
-        my @areas;
+        my @bodies = split( /,/, $problem{bodies_str} );
         my $cobrand = $problem{cobrand};
+
+        if (my $type = $cobrand_cls->call_hook(dashboard_problem_type => \%problem)) {
```

Comment:
I'm not sure I find this hook name very helpful. `categorize_problem_for_dashboard` or something like that would be a bit more helpful.